purchase_order_line_sequence/models/purchase.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Camptocamp SA - Damien Crier, Alexandre Fayolle
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+# Copyright 2017 Serpent Consulting Services Pvt. Ltd.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from openerp import api, fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    @api.multi
+    @api.depends('order_line')
+    def _compute_max_line_sequence(self):
+        """Allow to know the highest sequence entered in purchase order lines.
+        Then we add 1 to this value for the next sequence which is given
+        to the context of the o2m field in the view. So when we create a new
+        purchase order line, the sequence is automatically max_sequence + 1
+        """
+        for purchase in self:
+            purchase.max_line_sequence = (
+                max(purchase.mapped('order_line.sequence') or [0]) + 1)
+
+    max_line_sequence = fields.Integer(string='Max sequence in lines',
+                                       compute='_compute_max_line_sequence')
+
+    @api.multi
+    def _create_picking(self):
+        res = super(PurchaseOrder, self)._create_picking()
+        for order in self:
+            if any([ptype in ['product', 'consu'] for ptype in
+                    order.order_line.mapped('product_id.type')]):
+                picking = order.picking_ids.filtered(
+                    lambda x: x.state not in ('done', 'cancel'))[0]
+                for move, line in zip(
+                        sorted(picking.move_lines,
+                               key=lambda m: m.id), order.order_line):
+                    move.write({'sequence': line.sequence})
+        return res
+
+    @api.multi
+    def _reset_sequence(self):
+        for rec in self:
+            current_sequence = 1
+            for line in rec.order_line:
+                line.sequence = current_sequence
+                current_sequence += 1
+
+    @api.multi
+    def write(self, line_values):
+        res = super(PurchaseOrder, self).write(line_values)
+        self._reset_sequence()
+        return res
+
+    @api.multi
+    def copy(self, default=None):
+        return super(PurchaseOrder,
+                     self.with_context(keep_line_sequence=True)).copy(default)
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = 'purchase.order.line'
+    _order = 'sequence, id'
+
+    sequence = fields.Integer(help="Gives the sequence of the line when "
+                                   "displaying the purchase order.",
+                              default=9999)
+
+    sequence2 = fields.Integer(help="Displays the sequence of the line in "
+                                    "the purchase order.",
+                               related='sequence', readonly=True)
+
+    @api.multi
+    def _create_stock_moves(self, picking):
```

Comment:
In version 8.0, these method available at `purchase.order` object with more arguments. Here is reference https://github.com/odoo/odoo/blob/8.0/addons/purchase/purchase.py#L840

Possible to add these method in test case?

Reply:
Hello @bodedra ,
thanks for your note. I don't understand what you mentioned here. 
This is a method on the model `purchase.order.line`.
The method you linked is on the model `purchase.order`.
So both have the same name and different signatures but also the are on different objects.

Reply:
@mathiasfrancke In v8.0, we don't have same method in `purchase.order.line` compare to v10.0. But we have same method in `purchase.order` object level. 

Here is method references:

v8.0 - purchase.order - https://github.com/odoo/odoo/blob/8.0/addons/purchase/purchase.py#L840

v10.0 - purchase.order.line - https://github.com/odoo/odoo/blob/10.0/addons/purchase/models/purchase.py#L719

So we have make compatible code to work same in v8.0 & v.10.0

Reply:
Migrated missing method.